src/boot_O2/loadfragment2.c

```diff
@@ -51,10 +92,11 @@ void Load2_Relocate(void* allocatedVRamAddr, OverlayRelocationSection* ovl, uint
                 // Extract the address from the target field of the J-type MIPS instruction.
                 // Relocate the address and update the instruction.
 
-                *relocDataP =
-                    (*relocDataP & 0xFC000000) |
-                    ((RELOCATE_ADDR(PHYS_TO_K0((*relocDataP & 0x03FFFFFF) << 2), vRamStart, allocu32) & 0x0FFFFFFF) >>
-                     2);
+                if (1) {
+                    *relocDataP =
+                        (*relocDataP & 0xFC000000) |
+                        (((PHYS_TO_K0(MIPS_JUMP_TARGET(*relocDataP)) - vramStart + allocu32) & 0x0FFFFFFF) >> 2);
+                }
```

Comment:
Why the extra `if (1) {}`? From debug? or needed with the change in macros?

Reply:
This is actually a rather interesting question. Elliptic and I actually went on adventure trying to match as many different versions of this particular function (spanning from Yoshi Story, OoT, MM, and Animal Forest). MM's debug actually has a slightly refactored version (which is why these are not just copy paste from OoT, since the GC rom would have used the new refactored version), but it did indeed require an `if(1)` (our theory is it was meant to be something like the other two cases that actually relocate and test against `0x0F00000` but was typo'd to always evaluate to true).  On the other hand, Yoshi Story JP (which seems to be the very first iteration, and is also slightly different, but more similar to what MM has) also requires an `if(1)` around the same area, so also suggests the same idea.